Source/Scene/GltfLoader.js

```diff
@@ -0,0 +1,1132 @@
+import Cartesian3 from "../Core/Cartesian3.js";
+import Cartesian4 from "../Core/Cartesian4.js";
+import Check from "../Core/Check.js";
+import ComponentDatatype from "../Core/ComponentDatatype.js";
+import defaultValue from "../Core/defaultValue.js";
+import defined from "../Core/defined.js";
+import FeatureDetection from "../Core/FeatureDetection.js";
+import Matrix4 from "../Core/Matrix4.js";
+import Quaternion from "../Core/Quaternion.js";
+import Sampler from "../Renderer/Sampler.js";
+import getAccessorByteStride from "../ThirdParty/GltfPipeline/getAccessorByteStride.js";
+import getComponentReader from "../ThirdParty/GltfPipeline/getComponentReader.js";
+import numberOfComponentsForType from "../ThirdParty/GltfPipeline/numberOfComponentsForType.js";
+import when from "../ThirdParty/when.js";
+import AttributeType from "./AttributeType.js";
+import GltfFeatureMetadataLoader from "./GltfFeatureMetadataLoader.js";
+import GltfLoaderUtil from "./GltfLoaderUtil.js";
+import ModelComponents from "./ModelComponents.js";
+import ResourceCache from "./ResourceCache.js";
+import ResourceLoader from "./ResourceLoader.js";
+import ResourceLoaderState from "./ResourceLoaderState.js";
+import SupportedImageFormats from "./SupportedImageFormats.js";
+
+var Attribute = ModelComponents.Attribute;
+var Indices = ModelComponents.Indices;
+var FeatureIdAttribute = ModelComponents.FeatureIdAttribute;
+var FeatureIdTexture = ModelComponents.FeatureIdTexture;
+var MorphTarget = ModelComponents.MorphTarget;
+var Primitive = ModelComponents.Primitive;
+var Instances = ModelComponents.Instances;
+var Skin = ModelComponents.Skin;
+var Node = ModelComponents.Node;
+var Scene = ModelComponents.Scene;
+var Components = ModelComponents.Components;
+var Texture = ModelComponents.Texture;
+var MetallicRoughness = ModelComponents.MetallicRoughness;
+var SpecularGlossiness = ModelComponents.SpecularGlossiness;
+var Material = ModelComponents.Material;
```

Comment:
Question, given that we do use ES6 imports even though the rest of the code is ES5, would we be able to make ModelComponents a _module_ rather than a module + namespace object, then you we do:

```
import {
  Attribute,
  Indices,
  FeatureIdAttribute,
  //etc.
} from "./ModelComponents.js"
```

Though not sure if this is allowed in the current state of CesiumJS.

Reply:
I think this would most likely require changes to the build process because of how the Source/Cesium.js is generated. I'm not against the idea, but probably something that should be experimented/changed in a separate PR rather than trying to do as part of this one. Basically we need to think through what is means for end users.

What's weird to me is that all of these are shoved into a "ModelCompontent" object to begin with.  Why?

Reply:
@mramato thanks for the additional information. Any changes like that are definitely beyond the scope of this PR.

As for `ModelComponents`, these are small structs used only for storing parsed portions of the glTF JSON, they don't have any behavior. 

@lilleyse Hm... now that I think about it, does it make sense to have dedicated objects like this, or would it be enough to just use a regular JS object and document the properties via JSDoc's `@typedef`?

Reply:
One of the end goals is to allow users to build their own models so I think dedicated objects make sense. I haven't really nailed down the API yet though, and probably won't get to it until I start doing custom shaders. This might be an ok interim solution but separate files might make more sense.

Reply:
@lilleyse made a note for revisit this architecture when we do custom shaders. It doesn't need to hold up this PR.